client/blocks/reader-export-button/index.jsx

```diff
@@ -50,22 +50,26 @@ class ReaderExportButton extends React.Component {
 		}
 
 		if ( this.props.exportType === READER_EXPORT_TYPE_LIST ) {
-			wpcom.undocumented().exportReaderList( this.props.listId, this.onApiResponse );
+			exportReaderList( this.props.listId ).then( ( data ) => {
```

Comment:
You could change the onClick handler to be `async () =>` and then instead of using `.then()`, change it to be `const data = await exportReaderList(  this.props.listId )`. Then on the next line, you can check against the `isMounted` static property that you can introduce to this class, before running `this.onApiResponse( data )` (or the check could be inside `onApiResponse`).